openedx/features/course_experience/utils.py

```diff
@@ -118,26 +118,18 @@ def mark_last_accessed(user, course_key, block):
     course_key = CourseKey.from_string(course_id)
     course_usage_key = modulestore().make_course_usage_key(course_key)
 
-    if visual_progress_enabled(course_key=course_key):
-        # Deeper query for course tree traversing/marking complete
-        # and last completed block
-        block_types_filter = [
-            'course',
-            'chapter',
-            'sequential',
-            'vertical',
-            'html',
-            'problem',
-            'video',
-            'discussion'
-        ]
-    else:
-        # Shallower query is sufficient for last accessed block
-        block_types_filter = [
-            'course',
-            'chapter',
-            'sequential'
-        ]
+    # Deeper query for course tree traversing/marking complete
+    # and last completed block
+    block_types_filter = [
```

Comment:
Why take out the waffle switch code? I believe the waffle switch code is there because getting only 3 block types is much more performant.

Reply:
I had initially changed the waffle flag here to check for whether the course outline is enabled, because it has nothing to do with visual progress and only affects whether third-level blocks are visible in the course outline. After talking to the team at standup one morning, we decided just not to bother with adding in the check at all, since we would be removing it within a week or so, and if we are okay with this reduced performance post-updated course outline, it should be fine for the old course outline.